From 8dc1750cb2fe1800a82f451f35885da3f9fe7760 Mon Sep 17 00:00:00 2001
From: Peter Rugg <pdr32@cam.ac.uk>
Date: Tue, 7 Jan 2025 10:08:52 +0000
Subject: [PATCH] Fixes for running with TestRIG

Combination of the following commits: Capabilities-Limited/sail-cheri-riscv@8dc1750c,
Capabilities-Limited/sail-riscv@9479bf45

----

Fix resets

(cherry picked from rvy-sail-cheri-riscv commit 8dc1750cb2fe1800a82f451f35885da3f9fe7760)
(squashed with rvy-sail-cheri-riscv/sail-riscv commit 9479bf45a7661254668a2677f169797d726cf1af: Fix resets)
---
 model/cheri/cheri_step_ext.sail | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/model/cheri/cheri_step_ext.sail b/model/cheri/cheri_step_ext.sail
index f9a6dc2c..e92f6c91 100644
--- a/model/cheri/cheri_step_ext.sail
+++ b/model/cheri/cheri_step_ext.sail
@@ -37,7 +37,10 @@ function ext_reset_cheri() -> unit = {
   };
 }
 
-function ext_reset() -> unit = if get_config_rvfi() then ext_rvfi_reset() else ext_reset_cheri()
+function ext_reset() -> unit = {
+  ext_reset_cheri();
+  if get_config_rvfi() then ext_rvfi_reset();
+}
 
 function ext_reset_misa() -> unit = ()
 
-- 
2.56.0
